ERCS/erc-7844.md

```diff
@@ -0,0 +1,391 @@
+
+
+## Backwards Compatibility  
+
+This ERC introduces a new design pattern and does not interfere with existing Solidity implementations.  CDS *does* *not* implicitly interfere with common libraries such as those provided by OpenZeppelin, but is not supported explicitly. Library-imposed global data within CDS-linked contracts can be a problem.  The ideal approach is to use lightweight libraries and to avoid mapped data in CDS-linked contracts whenever possible.
+
+## **Test Cases**
+
+### **1. Core Functionality**
+
+- **Initialization**
+    - Input: `types = [1, 3, 6], sizes = [32, 8, 128]`.
+    	- Expected: Storage space initialized with 3 members.
+- **Insert New Members**
+    - Input: `insert_new_member(1, 128, storageSpace = 0)`.
+    	- Expected: New `uint128` member added with correct `bitCount`.
+- **Data Storage and Retrieval**
+    - Input: `put(42, memberIndex = 0, entryIndex = 0, storageSpace = 0)` → `get(0, 0, 0)`.
+    	- Expected: `42`.
+
+### **2. Edge Cases**
+
+- **String Handling**
+    - Input: Insert five strings consecutively.
+    	- Expected: No collisions; strings retrieved accurately.
+    
+    - Input: insert two dynamic strings, then one uint256
+		- Expected: uint256 is properly configured with:
+				`bitCount == 0` 
+			because: 
+				`safeIndex == 0` maps to the dynamic string with index 0.  This zero value fills both decoded `{type, size}` slots in the standard type construction logic.  Hence, we begin with a valid bitCount of 0.
+		    
+- **Entry Creation**
+    - Input: Add 10,000 entries to a storage space with `pushMany`.
+    	- Expected: System can store to  any of these entry indices.
+    
+- **Invalid Input**
+    - Input: `put(42, memberIndex = 1, storageSpace = 0)`.
+    	- Expected: Reverts with error.
+    
+	* Input: `put("42", memberIndex = 1, entryIndex = 0, storageSpace = 0)`.
+    	- Expected: Reverts with error.
+    
+	- Input: `put_string(42, memberIndex = 1, entryIndex = 0, storageSpace = 0)`.
+    	- Expected: Reverts with error.
+
+## Gas Benchmarks
+
+* This section assumes that storage operations interact with pre-populated slots.  
+* These values are derived from the optimized HoneyBadger model and serve as good targets for efficiency.  
+* Displayed values reference *execution* cost.  
+
+	`init_create([1], [256]):` 93,860 gas  
+	`init_create([1,1,1], [8,256,256]):` 140,024 gas  
+	`insert_new_member:` 40,653 gas  
+	`push:` 10,316 gas  
+	`put:` 15,543 gas  
+	`put_batch([20, 20], [0, 1], 0, 0):` 22,895 gas  
+	`get`: 9374 gas  
+
+
+## Reference Implementation
+
+Refer to [CDS Minimal Example](https://github.com/wisecameron/ConsolidatedDynamicStorage/blob/main/contracts/CDSMinimal.sol)
```

Comment:
Please either include your reference implementation in the `assets/...` directory or remove this section. We do not permit external links except for those listed in EIP-1.

Reply:
Thank you for taking the time to dig through and find these issues -- will fix each as soon as possible.